python/glow/gwas/log_reg.py

```diff
@@ -182,20 +197,48 @@ def _logistic_regression_inner(genotype_pdf: pd.DataFrame, log_reg_state: LogReg
     c, d: covariate
     '''
     X = np.column_stack(genotype_pdf[_VALUES_COLUMN_NAME].array)
+
+    # Based on Joelle's code
```

Comment:
Did you try this? I see that Joelle pointed out that in regenie they actually only use the result of this residualization for the firth correction.